src/runtime/dso_library.cc

```diff
@@ -136,6 +166,50 @@ void DSOLibrary::Load(const std::string& name) {
 
 void* DSOLibrary::GetSymbol_(const char* name) { return dlsym(lib_handle_, name); }
 
+void DSOLibrary::ListSymbols_(std::function<void(const char*)> callback) {
+  // Adapted from https://stackoverflow.com/a/62205128
+
+  struct link_map* map = nullptr;
+  dlinfo(lib_handle_, RTLD_DI_LINKMAP, &map);
+
+  Elf64_Sym* symbol_table = nullptr;
```

Comment:
I am not sure if this will work on windows

Reply:
This particular implementation definitely wouldn't work on Windows, and is in the `#ifdef` for the linux-specific code.  Currently, the Windows implementation returns an empty list.